### PR TITLE
[TTAHUB-2977] Add first version of CDI grant filter

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

This will be a bit scaled back from what the ticket initially asked for as it overlaps with the work Jon is currently doing for grants. And the QA work that Garrett is completing for the SSDI.

For now this filter will be purely on the CDI column on the grants table (no associated grants). It will also be currently excluded from the QA dashboard as the page is still a WIP. 

**Note:** When filtering inactive grants on the RTR search page we by default limit the possible inactive grants using the below query. You will need to account for this when matching to what's in the db via a query.
`("grants"."endDate" > '2020-08-31 04:00:00.000 +00:00' AND ("grants"."inactivationDate" IS NULL 
OR "grants"."inactivationDate" > '2020-08-31 04:00:00.000 +00:00'))`

## How to test

- Review the code.
- Make sure the grant filter works as expected on the various pages (see ticket)

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-2977

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/components/filter/activityReportFilters.js [hunk 1]
```diff
@@ -347,6 +348,24 @@ export const ttaTypeFilter = {
...
 };
 
+export const cdiGrantsFilter = {
+  id: 'cdiGrants',
+  display: 'CDI grants',
+  conditions: FILTER_CONDITIONS,
+  defaultValues: {
+    is: 'active',
+    'is not': 'active',
+  },
```
Comment (thewatermethod) [inline frontend/src/components/filter/activityReportFilters.js:358]: AC asks for the following:

Grant status
-     is / is not
-     Active / Inactive / Interim management (CDI)

Are we not doing that?

Comment (AdamAdHocTeam) [reply frontend/src/components/filter/activityReportFilters.js:358]: Aren't these just the default values we show?

Comment (thewatermethod) [reply frontend/src/components/filter/activityReportFilters.js:358]: "Grant status" not "CDI Grants" to be clear

File: src/scopes/activityReport/grantStatus.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import { sequelize } from '../../models';
+
+const grantStatusSql = (grantStatus, notWithin) => {
+  let where = null;
+  if (grantStatus === 'active') {
+    where = `"Grants"."status" = '${notWithin ? 'Inactive' : 'Active'}' AND "Grants"."cdi" = false`;
+  } else if (grantStatus === 'inactive') {
+    where = `"Grants"."status" = '${notWithin ? 'Active' : 'Inactive'}' AND "Grants"."cdi" = false`;
+  } else {
+    where = `"Grants"."cdi" = ${notWithin ? 'false' : 'true  AND "Grants"."status" = \'Active\''}`;
```
Comment (GarrettEHill) [inline src/scopes/activityReport/grantStatus.js:11]: I don't think this case is correct when not within is true.

File: src/scopes/grants/grantStatus.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import { sequelize } from '../../models';
+
+const grantStatusSql = (grantStatus, notWithin) => {
+  let where = null;
+  if (grantStatus === 'active') {
+    where = `"Grants"."status" = '${notWithin ? 'Inactive' : 'Active'}' AND "Grants"."cdi" = false`;
+  } else if (grantStatus === 'inactive') {
+    where = `"Grants"."status" = '${notWithin ? 'Active' : 'Inactive'}' AND "Grants"."cdi" = false`;
+  } else {
+    where = `"Grants"."cdi" = ${notWithin ? 'false' : 'true  AND "Grants"."status" = \'Active\''}`;
```
Comment (GarrettEHill) [inline src/scopes/grants/grantStatus.js:11]: Same as above

File: frontend/src/components/filter/grantFilters.js [hunk 4]
```diff
@@ -74,3 +75,21 @@ export const recipientsWithoutTTA = {
     />
   ),
 };
+
+export const cdiGrantsFilter = {
```
Comment (thewatermethod) [inline frontend/src/components/filter/grantFilters.js:79]: we happy with this name still?

**Review by thewatermethod — APPROVED** [review]
Looks good to me, seems to work, but please confirm with @GarrettEHill as he raises a concern below

File: frontend/src/components/filter/grantFilters.js (hunk 4 above)
Comment (AdamAdHocTeam) [reply frontend/src/components/filter/grantFilters.js]: Updated.

File: src/scopes/activityReport/grantStatus.js (hunk 2 above)
Comment (AdamAdHocTeam) [reply src/scopes/activityReport/grantStatus.js:11]: Ok I spoke with Kelly for now only show we will keep as is with active, inactive, and active cdi.

File: src/scopes/grants/grantStatus.js (hunk 3 above)
Comment (AdamAdHocTeam) [reply src/scopes/grants/grantStatus.js:11]: Ok I spoke with Kelly for now only show we will keep as is with active, inactive, and active cdi.

**Pull request merged**